### PR TITLE
updated seeder and added filter by platform

### PR DESCRIPTION
**Pull request by Jovialiste82**
Hey guys, i struggled a bit with async behaviour while looping through an array is involved but i think the filtering by platform works now. Can you pls have a look?

### CODE REVIEW

File: server/controllers/mentionController.js [hunk 1]
```diff
@@ -2,12 +2,35 @@
...
+
+    // Fetching Mentions pertaining to selected platforms
+    const fetchMentions = async (array) => {
+      const allMentions = [];
+      for (const platform of array) {
+        const mentions = await Mention.find({ platform });
+        allMentions.push(mentions);
+      }
+      return allMentions;
+    };
```
Comment (dproc96) [inline server/controllers/mentionController.js]: You can optimize this by handling the promises in parallel. Check out the docs on [Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

File: server/controllers/mentionController.js [hunk 2]
```diff
@@ -2,12 +2,35 @@
...
+
+    const result = await fetchMentions(platformsArray);
+    const allMentions = result.flat();
+
+    // TODO : filter by keyword  (scraping library or algorithm ?)
+    // TODO : sort by sortbydate
+    // TODO : sort by sortbypop
+
+    res.json(allMentions);
```
Comment (dproc96) [inline server/controllers/mentionController.js]: The structure of this json is strange, would be better to return an object structured like `{ mentions: allMentions }`

File: server/data/mentions.js [hunk 3]
```diff
@@ -3,61 +3,85 @@
...
+    content: 'Ut enim ad minim bitcoin veniam, quis nostrud exercitation ullamco JupiterCorp laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure MercuryCorp dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
+    platform: "twitter",
+    image: 'https://upload.wikimedia.org/wikipedia/fr/f/fc/Reddit-alien.png',
+    date: new Date('May 15, 2020 16:27:27'),
+    popularity: 902,
+  },
 
 ]
 
-module.exports = mentions
```
Comment (dproc96) [inline server/data/mentions.js:63]: Why did you remove the export for this file?

**Review by dproc96 — COMMENTED** [review]
I think I see what the problem you encountered with the multiple async tasks, see my comment below

File: server/controllers/mentionController.js (hunk 1 above)
Comment (Jovialiste82) [reply server/controllers/mentionController.js]: I just changed the code to use Promise.all ... were you referring to that option ?

File: server/controllers/mentionController.js (hunk 2 above)
Comment (Jovialiste82) [reply server/controllers/mentionController.js]: Done ! Thx !

File: server/data/mentions.js (hunk 3 above)
Comment (Jovialiste82) [reply server/data/mentions.js:63]: Actually it's showing on line 87. I did add some new Mentions in the file.

**Comment by Jovialiste82** [comment]
@dproc96  I just finished the whole "filter/sort" code. Can you pls double-check ?

File: server/controllers/mentionController.js [hunk 4]
```diff
@@ -1,17 +1,78 @@
-// const asyncHandler = require("express-async-handler");
 const Mention = require("../models/mentionModel");
 
 // @desc    Get Mentions
-// @route   GET /api/mentions/
+// @route   GET /api/mentions?platforms=xxx&keyword=xxx&sort=xxx
```
Comment (jessdesr) [inline server/controllers/mentionController.js]: Take a look at JSDoc @param, you should be able to define the individual query parameters instead of including them in the route like this

File: server/controllers/mentionController.js [hunk 5]
```diff
@@ -1,17 +1,78 @@
...
+            filteredMentions.push(mention);
+          }
+        });
+      }
+
+      // sort by selector
+      let sortedMentions = [];
+
+      switch (sort) {
+        case "popularity":
```
Comment (jessdesr) [inline server/controllers/mentionController.js:54]: I would probably sort within the Mongo query instead of sorting the results afterwards, would probably be more efficient performance-wise.  Maybe it's something we test later on?

File: server/controllers/mentionController.js (hunk 4 above)
Comment (Jovialiste82) [reply server/controllers/mentionController.js]: @jessdesr  First time ever I use this, so not too sure if I got it right

File: server/controllers/mentionController.js (hunk 5 above)
Comment (Jovialiste82) [reply server/controllers/mentionController.js:54]: @jessdesr well, the user is able to toggle the switch button from the UI, so that's something I would think would need to be handled by an API call to the back-end ?

**Review by jessdesr — APPROVED** [review]
